core/services/ardupilot_manager/mavlink_proxy/Endpoint.py

```diff
@@ -1,76 +1,56 @@
-        (["file"], EndpointType.File),
-    ]
-
-    def __init__(self, constructor: Union[str, Dict[str, str]] = "") -> None:
-        if isinstance(constructor, dict):
-            self.from_dict(constructor)
-        else:
-            self.from_str(constructor)
-
-    def from_str(self, pattern: str) -> "Endpoint":
-        args = pattern.split(":")
-        if len(args) != 2 and len(args) != 3:
-            raise RuntimeError("Wrong format for endpoint creation.")
-
-        for stringType, endpointType in self._connTypeMap:
-            if args[0] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
 
-        self.place = args[1]
-        self.argument = args[2] if len(args) == 3 else None
+class EndpointType(str, Enum):
+    UDPServer = "udpin"
+    UDPClient = "udpout"
+    TCP = "tcp"
+    Serial = "serial"
+    File = "file"
 
-        return self
 
-    def from_dict(self, pattern: Dict[str, str]) -> "Endpoint":
-        if not {"connType", "place"}.issubset(pattern.keys()):
-            diff = {"connType", "place"}.difference(pattern.keys())
-            raise RuntimeError(f"Wrong format for endpoint creation, missing the following: {diff}.")
-
-        for stringType, endpointType in self._connTypeMap:
-            if pattern["connType"] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
-
-        self.place = pattern["place"]
-        self.argument = pattern["argument"] if pattern["argument"] else None
-
-        return self
+@dataclasses.dataclass
+class Endpoint:
+    connection_type: str
+    place: str
+    argument: int
+
+    @root_validator
+    @classmethod
+    def is_mavlink_endpoint(cls: Type["Endpoint"], values: Any) -> Any:
+        connection_type, place, argument = (values.get("connection_type"), values.get("place"), values.get("argument"))
+        if connection_type in [EndpointType.UDPServer, EndpointType.UDPClient, EndpointType.TCP]:
+            if validators.domain(place) or validators.ipv4(place) or validators.ipv6(place):
+                if argument in range(1, 65536):
+                    return values
+                raise ValueError(f"Ports must be in the range 1:65535. Received {argument}.")
+            raise ValueError(f"Invalid network address: {place}")
+
+        if connection_type == EndpointType.Serial.value:
+            if is_valid_filepath(place):
+                if argument in range(0, 10000001):
+                    return values
+                raise ValueError(f"Serial baudrate must be in the range 0:10000000. Received {argument}.")
+            raise ValueError(f"Bad serial address: {place}.\nMake sure to use an absolute path.")
+
+        if connection_type == EndpointType.File.value:
+            if is_valid_filepath(place):
+                if argument in range(-1, 101):
```

Comment:
What is argument for file endpoint ?

Reply:
Stream rate, or I got it wrong?

Reply:
a file endpoint is just a place where we are going to store any incoming message, stream rate is not an option that I can see to be used and how to be used.

Reply:
Mistook it with the idea of a log file again.